### PR TITLE
Merge `riganti/dotvvm-tracing` into this repo

### PR DESCRIPTION
**Pull request by cafour**

### CODE REVIEW

File: src/Tools/publish.ps1 [hunk 1]
```diff
@@ -0,0 +1,153 @@
+param([String]$version, [String]$apiKey, [String]$server, [String]$branchName, [String]$repoUrl, [String]$nugetRestoreAltSource = "", [bool]$pushTag, [String]$configuration, [String]$apiKeyInternal, [String]$internalServer, [String]$signUser = "", [String]$signSecret = "", $signConfigPath = "")
```
Comment (quigamdev) [inline src/Tools/publish.ps1]: All powershell scripts should be in `ci\scripts`. Just do not know how to differentiate them from framework scripts. What about adding folders or names of packages to the name of scripts?

**Comment by quigamdev** [comment]
Please remove `src/Tools/nuget.exe` since there is nuget.exe in `ci\scripts` we do not need another one there.

**Comment by tomasherceg** [comment]
Few quick thoughts:
1. Do we want the samples for tracing to be in the root of the repo? Wouldn't it be better to have them in `src/Tracing/samples` or in `src/Samples/Tracing`?
2. I noticed that in the new structure we don't have the `src\Tools\build\publish.ps1` script which is used when NuGet packages are published. I cannot find the script anywhere, and the paths to the packages added by this PR should be added there.

**Comment by cafour** [comment]
@tomasherceg

1. I actually just forgot about the samples. I moved them to `/src/Samples`.

2. I'm not sure how publish is being handled in the new structure. (@quigamdev ?)

File: src/Tracing/ApplicationInsights/ApplicationInsightsBulkTracer.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        public ApplicationInsightsBulkTracer(TelemetryClient telemetryClient)
+        {
+            this.telemetryClient = telemetryClient;
+        }
+
+        public Task EndRequest(IDotvvmRequestContext context)
+        {
+            foreach (var @event in events)
+            {
+                telemetryClient.TrackMetric(@event.Name, @event.TimeStamp);
```
[the remark this answers is not in the document]
Comment (cafour) [reply src/Tracing/ApplicationInsights/ApplicationInsightsBulkTracer.cs]: You're right, I dived into the `microsoft/ApplicationInsights-dotnet` repo and found a buffer in `ServerTelemetryChannel`.

**Review by exyi — APPROVED** [review]

**Review by quigamdev — APPROVED** [review]

**Pull request merged**